src/main/rx/spektrum.c

```diff
@@ -261,10 +261,12 @@ const uint8_t vtxTrampPi[] = {           // Spektrum Spec    Tx menu  Tx sends
 #ifdef USE_VTX_COMMON_FREQ_API
       uint16_t freq = SpektrumVtxfrequencyTable[vtx.band][vtx.channel];
       uint16_t currentFreq = 0;
-      vtxCommonGetFrequency(&currentFreq);
-      if (currentFreq != freq) {
-        vtxCommonSetBandAndChannel(VTX_COMMON_BAND_USER, vtx.channel);
-        vtxCommonSetFrequency(freq);
+      if (vtxCommonDeviceRegistered()) {
+        vtxCommonGetFrequency(&currentFreq);
```

Comment:
Same here, any reasons to not just touch the configuration here?

Reply:
Still trying to work out the 'save' problem.  Using the parameter group requires a eeprom write/read, which is a generally expensive process and should not be done rapidly for small changes.  Usually, these writes/reads are best handled through user action instead of implicitly.

Reply:
Ok, got it. You probably already thought about buffering the changes and writing them after a given delay.

Reply:
ideally, the different extensions would explicitly request the save.  The only one that doesn't do that right now is the spectrum piece.

I've also made a change to persist settings to the vtx based on the mutable configuration rather than the committed configuration.  So this will help.  Just need to have a 'save' option on the spektrum menu to persist the change.

Reply:
I've thought about implicit periodic persistence checks against the eeprom (writing after a delay) but this could elicit unwanted 'hiccups' on the fc, as the writes will happen without the user expecting it.  Also, if someone makes a lot of changes, it could choke on the writes and appear unresponsive or raise the same issue as found in #4437

Reply:
I guess one of my main missions with this was to remove the need for implicit writes.